documents/xaml-suggested-actions/xaml-designer-suggested-actions-extensibility.md

```diff
@@ -0,0 +1,110 @@
+## Suggested Actions Extensibility
+
+- [Tokens](./xaml-designer-suggested-actions-extensibility-tokens.md)
+- [Features](./xaml-designer-suggested-actions-extensibility-features.md)
+- [Behaviors](./xaml-designer-suggested-actions-extensibility-behaviors.md)
+- [Customization](./xaml-designer-suggested-actions-extensibility-customization.md)
+
+Starting from Visual Studio 2019 16.7 Preview 3 we added extensibility support for "XAML Suggested Actions"
+>*Note: this API will be available only in Preview Channel until feature is finalized and completed*
+
+To enable "Xaml Suggested Actions" for any control, `SuggestedActionProvider` feature provider should be created and registered in metadata.
+
+>*Control could have multiple Suggested Actions providers. Each separate provider will be shown as a separate tab in UI.*
+
+Example for Button:
+
+![extensibility-migration-architecture](xaml-suggested-actions.png)
+
+### Implementation for example above:
+```CS
+public class ButtonActionProvider : SuggestedActionProvider
+{
+    public static ActionToken Token_Group_Common        = SuggestedActionProviderTokens.Token_Group_Common;
+    public static ActionToken Token_Group_Specific      = SuggestedActionProviderTokens.Token_Group_Specific;
+    public static ActionToken Token_Property_Background = SuggestedActionProviderTokens.Token_Property_Background;
+    public static ActionToken Token_Property_Content    = SuggestedActionProviderTokens.Token_Property_Content;
+    public static ActionToken Token_Property_IsEnabled  = SuggestedActionProviderTokens.Token_Property_IsEnabled;
+    public static ActionToken Token_Property_IsCancel   = new ActionToken(0x1001);
+    public static ActionToken Token_Property_IsDefault  = new ActionToken(0x1002);
+    public static ActionToken Token_Last = new ActionToken(0x10FF);
+    
+    public override string Header { get => "Actions"; }
+    public override string Type { get => "System.Windows.Controls.Button"; }
+    
+    public override void Initialize()
+    {
+        this.ShowNameProperty = true;
+        base.Initialize();
+        this.AddGroup(new ActionGroup(ButtonActionProvider.Token_Group_Common,
+            new PropertyAction(ButtonActionProvider.Token_Property_Content, "Content"),
+            new PropertyAction(ButtonActionProvider.Token_Property_Background, "Background"),
+            new PropertyAction(ButtonActionProvider.Token_Property_IsEnabled, "IsEnabled")
+            )); ;
+        this.AddGroup(new ActionGroup(ButtonActionProvider.Token_Group_Specific,
+            new PropertyAction(ButtonActionProvider.Token_Property_IsCancel, "IsCancel"),
+            new PropertyAction(ButtonActionProvider.Token_Property_IsDefault, "IsDefault")));
+        this.AddGroup(new BorderBrushActionGroup());
+        this.AddGroup(new VisibilityActionGroup());
+        this.AddGroup(new FontSettingsActionGroup());
+    }
+}
+```
+### Example for `MyButtonSuggestedActionProvider : ButtonActionProvider`
```

Comment:
I'm not sure what this means. It seems unrelated to the image above? Seems like this needs more explanation.